pkg/test/util/connection/connection.go

```diff
@@ -21,13 +21,21 @@ import (
 	"istio.io/istio/pkg/test/framework/components/apps"
 )
 
+const (
+	AllowHTTPRespCode = "200"
+	DenyHTTPRespCode  = "403"
+	TCPPort           = 90
+)
+
 type Connection struct {
 	From            apps.KubeApp
 	To              apps.KubeApp
 	Protocol        apps.AppProtocol
 	Port            int
 	Path            string
 	ExpectedSuccess bool
+	// Whether the request is allowed (the expected value could be 200 or 403), used for RBAC tests.
+	IsAllowed bool
```

Comment:
You have moved the `checkRBACRequest()` out of connection.go so there is no need to keep the `IsAllowed` in connection.go otherwise you're only doing half of the change. The point is the connection.go itself should be test case agnostic, the `IsAllowed` is only used to pass around the expected value between `TestRBACV2()` and `checkRBACRequest()`, it's not even used in `connectin.go`.

Reply:
The reason I keep it there is because I think `IsAllowed` is part of a request/connection response. Plus, the test is using `Connection` struct (on going discussion https://github.com/istio/istio/pull/13384#discussion_r276798449), so I think it should belong here.

@lei-tang wrote this file for some security tests, do you have any comments?

Reply:
If the semantic of "IsAllowed" is to represent whether the requests of a connection are allowed, I suggest renaming it to "IsRequestAllowed" to be more clear about its meaning. Meanwhile, I suggest changing the comment of this field from "// Whether the request is allowed (the expected value could be 200 or 403), used for RBAC tests" to "// Whether the requests of the connection are allowed." In addition, please set the default value of this field carefully to avoid wrong test results.

Reply:
`IsAllowed` should be decoupled with `Connection`. It's not related to `Connection` but used only as a way to pass the expected value from `TestRBACV2()` to `checkRBACRequest()`.

Reply:
I think `Connection` here can be used interchangeably/read as Request. Do you want to change the name Connection to Request? Then IsAllowed is definitely part of a request.

Reply:
Given how this structure is used in the tests, Request is more accurate than Connection.

Reply:
Are you okay with this? @yangminzhu

Reply:
I'm not against with the name, either `Request` or `Connection` is fine to me. I'm concerned that putting RBAC specific logic to the test framework. Specifically here, the `Connection` (or `Request`) is about the request being sent to the server which is handled by the test framework, On the other side, the expected result (i.e. `IsAllowed`) is another thing that is only cared by the specific test case (i.e. `TestRBACV2()`)

Also what's the relationship between this `IsAllowed` and `ExpectedSuccess`? It's hard to understand the difference between them.

Reply:
`Connection` is not necessarily tied to the test framework. We could have just put it in `test/intergration/security`. @lei-tang wrote it. Only security tests are using it now. I don't know if people from other team are gonna use it. Since the `Call` function has been modified and callers are expected to handle error checking on their own, I suspect not, because their use case could be different than ours.

`IsRequestAllowed` and `ExpectedSuccess` are pretty the same (for now). I'm using `IsRequestAllowed` specifically since it makes more sense in RBAC. `ExpectedSuccess` does not tell us if success here means getting 200 or 403.

Reply:
At least merge the `IsRequestAllowed` and `ExpectedSuccess` to a single thing. Otherwise, if I add some other tests in the future, I may add another thing similar to the `IsRequestAllowed`/`ExpectedSuccess` but with some minor difference as it makes more sense in my test, so every test case is adding a specific expectation to this struct as it makes more sense for their specific use case.